### PR TITLE
refactor(libs): remove libs barrels and use direct imports

### LINKED ISSUE #1727: Phase: Remove the `@/libs` barrel; migrate to direct imports + `@libs/*` semantic aliases
### Summary

As a **standalone phase** of the [barrel-removal epic](https://github.com/pubky/pubky-app/issues/1722), **retire the `src/libs` aggregate barrel** (`import * as Libs from '@/libs'` and the `export *` chain in `src/libs/index.ts`). Replace with **direct, named imports** from the real modules under `src/libs/…`, using **short semantic path aliases** (e.g. `@libs/utils/...`, `@libs/http/...`) so imports stay readable without going back to a fat barrel.

This phase is **only** about **`src/libs`**, not `@/core`, components, or hooks.

### Why this phase

- **`libs` is a clear vertical:** shared helpers, HTTP, errors, env, etc.—a good first slice after aliases land.
- **Smaller final migration:** a lot of `Libs.*` usage and `vi.mock('@/libs', …)` can be fixed **before** the last big push on core and UI.
- **Tests:** today many files treat `@/libs` as one blob. After this, mocks and imports can target **real submodules**, so tests stay accurate when the rest of the repo moves off barrels.
- **Shorter imports without a barrel:** semantic aliases (e.g. `@libs/*` → `./src/libs/*`) keep paths **short and stable** while still importing from **concrete files**—no need for a giant `index.ts` re-export.

### Current state (typical)

- `src/libs/index.ts` re-exports sub-areas with `export * from './api' | './http' | './utils' | …` (exact list in repo).
- Call sites use **namespace** imports: `import * as Libs from '@/libs'`, then `Libs.cn`, `Libs.postJson`, etc.

### Target state

- **No** aggregate barrel at `src/libs` that re-exports the whole tree for convenience (per **#1722**: barrel files that only exist to aggregate internal modules **go**).
- Call sites use **named imports** from the **defining module**, with paths expressed through **`@libs/...` aliases** (or the project’s chosen prefix) so file paths are **short** and **explicit**.
- **Documentation** updated (see below) so everyone uses the same import style for `libs` after this phase.

### Cross-layer imports (explicitly allowed for now)

- If some **`src/libs`** code still imports from a **`@/core` barrel** (or other non-`libs` barrel), that is **acceptable in this phase**. Those barrels will be removed in **later** epic work; **do not** expand scope to “fix all of `core`” here. Prefer **not** adding *new* tight coupling where avoidable; document anything that must stay until the core phase.

### Documentation (required)

- Update **contributor-facing docs** (e.g. `AGENTS.md`, component / architecture notes, or a short section in `docs/`) to describe:
  - **How** to import from `libs` after this change (semantic aliases + direct module paths).
  - **That** `src/libs` no longer exposes a single catch-all barrel.
  - **That** temporary imports from `@/core` (or similar) from within `libs` may still exist until the **core** migration runs.

### Suggested approach

1. **Prerequisite:** `tsconfig` (and **Vitest** / e2e tooling, if applicable) includes **`@libs/*`** (or equivalent) mapping to `./src/libs/*`, **alongside** existing paths—so migration can be incremental.  
2. **Rewrite call sites:** replace `import * as Libs` + `Libs.foo` with **direct imports** using the **semantic alias** to the right file.  
3. **Remove** the aggregating `src/libs/index.ts` re-exports.  
4. **Tests:** update `vi.mock` / imports that target `@/libs` as a monolith so they match **concrete** `libs` modules where practical.  
5. **Validate:** `tsc`, ESLint, unit tests, production build; spot-check critical flows.  
6. **Docs:** apply the documentation updates above.

### Acceptance criteria

- [ ] No **`import * as Libs from '@/libs'`** (or equivalent “whole `libs` barrel” pattern) in normal app or test code, except any **briefly documented** edge case.  
- [ ] **`src/libs/index.ts`** no longer **aggregates** the public `libs` API via `export *` (barrel **removed** per **#1722**).  
- [ ] **Semantic aliases** for `libs` are in place and used so imports are **short** and point to **real files** (not a re-export hub).  
- [ ] **Docs** updated per “Documentation (required)”.  
- [ ] **CI green:** typecheck, lint, unit tests, build.  
- [ ] **Cross-layer:** any remaining `libs` → `@/core` (etc.) usage is **acknowledged** as **temporary** until the core phase; no requirement to remove `@/core` barrels **in this issue**.

### Out of scope

- Removing barrels under **`src/core`**, **`src/components`**, **`src/hooks`**, etc. (other phases of **#1722**).  
- **Runtime behavior** changes to helpers (import-only refactor, aside from bugfixes from stricter types/tests).  
- **Icons / any separate asset pipeline**—assumed **not** part of this `libs` barrel; handle independently if still applicable elsewhere.

### Links

- **Parent:** [#1722 — Epic: remove barrel imports](https://github.com/pubky/pubky-app/issues/1722)

### PR DESCRIPTION
**Pull request by infin1t3**
## Summary
- Removes the `src/libs` aggregate barrel and nested libs `index.ts` re-export files.
- Migrates app, tests, and docs to direct `@/libs/...` file imports.
- Syncs the branch with latest `dev` and keeps the existing `@/icons` alias convention.

## Validation
- `npm run format:check`
- `npm run lint`
- `npm run typecheck`
- `npm run test`

## Testing
Basic regression testing done

https://github.com/user-attachments/assets/a74b2fc7-bf61-4e6a-b517-e9695d039e72

Resolves #1727

### CODE REVIEW

File: src/components/atoms/Input/Input.tsx [hunk 1]
```diff
@@ -17,9 +16,9 @@ const defaultProps = {
...
   const { theme } = { ...defaultProps, ...props };
 
-  const inputClassName = Libs.cn(
-    'file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50',
-    'aria-invalid:ring-destructive/20 aria-invalid:ring-destructive/40 aria-invalid:border-destructive focus:outline-none hover:outline-none hover:ring-0 focus:ring-0 focus:ring-offset-0',
+  const inputClassName = cn(
+    'flex h-9 w-full min-w-0 rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none selection:bg-primary selection:text-primary-foreground file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50',
```
Comment (infin1t3) [inline src/components/atoms/Input/Input.tsx:20]: This is just a reorder

**Review by Taewa — APPROVED** [review]
LGTM

Since there are 593 files changed I didn't check files one by one. And I check the following ways

## 1. Manual global search
Make sure none of the below exist on the codebase.
- `Libs.`
- `* as Libs`
- `'@/libs'`

## 2. Manual app testing on local
Run the app and check UI, functionalities.
- Sanity testing on my local machine with `npm run dev`
- Sanity testing on my local machine with `npm run build && npm run start`


https://github.com/user-attachments/assets/c784e248-100a-4900-a55e-1c00a0737c17





## 3. Validation commands
```
npm run format:check
npm run lint
npm run typecheck
npm run test
```

## 4. Check with AI
Implementation correctness:
  - 0 remaining src/libs/**/index.ts barrel files.
  - 0 remaining from '@/libs' aggregate-barrel imports.
  - tsconfig.json drops the @/libs alias, keeps only @/libs/* and the documented @/icons exception (→
  src/libs/icons/icons.tsx).
  - ~30 distinct @/libs/* import targets, all pointing at concrete files.
  - Icon convention from AGENTS.md / .greptile/rules.md upheld: 151 direct lucide-react imports, 15 @/icons imports,
  urlToIcon helpers from @/libs/utils/urlToIcon.
  - 20-commit history is coherent (icons split → alias removals → test rewrites using real helpers → docs → cleanup),
  with two fix(libs): commits showing regressions were caught and preserved before merge.

  Verdict: properly implemented and ready for review. Nothing blocking found.
